jekyll/_ccie/high-availability.md

```diff
@@ -0,0 +1,280 @@
+    rs.add('<third hostname>')
+    ```
+    The MongoDB replica set is ready for use with CircleCI.
+
+7. Create a CircleCI user for the actual application, which can be created with: 
+    ```
+    db.createUser({ user: "circle", pwd: '',
+                    roles: [ { role: "readWrite", db: "circle_ghe" },
+                             { role: "readWrite", db: "build_state_dev_ghe" },
+                             { role: "readWrite", db: "containers_dev_ghe" } ] })
+    ```
+
+### MongoDB Backups
+
+This section describes how to backup data mounted on EBS volumes.
+
+1. To ensure that the disk is in a consistent state, stop the mongodb process (using `sudo service mongod stop` or another system-appropriate command) on one of the SECONDARY instances and wait for the process to completely stop. **NOTE:** stopping the replica outright has proven to be more reliable for consistent restores than using the db.fsyncLock() mechanism described in the mongo documentation. This is also an additional safeguard for consistent state on top of the journal files.
+2. Use the AWS console or CLI to generate and complete a snapshot.
+3. To rejoin the replica set as a SECONDARY, restart the mongodb  process with `sudo service mongod start`.
+4. The SECONDARY begins serving traffic after replication catches up. 
+
+## Administering PostgreSQL
+
+It is assumed that users of the initial high-availability offering will use RDS on AWS, with multi-AZ mode and automated backups. Please let us know if any additional information is required for configuring and maintaining Postgres.
+
+## High Availability Configuration
+
+Making the services box automatically recover from failure means replacing it with an ASG of a single member, where the associated userdata entirely specifies how to install and configure Replicated and connect to the external databases.
+
+A working example of this can be found at [https://github.com/circleci/enterprise-setup/compare/ha-test](https://github.com/circleci/enterprise-setup/compare/ha-test). Note in particular the userdata for the services box launch configuration: 
+
+```
+#!/bin/bash
+
+set -ex
+
+startup() {
+  apt-get update; apt-get install -y python-pip
+  pip install awscli
+  aws s3 cp s3://ha-test-bucket-3f5b105a/settings.conf /etc/settings.conf
+  aws s3 cp s3://ha-test-bucket-3f5b105a/replicated.conf /etc/replicated.conf
+  aws s3 cp s3://ha-test-bucket-3f5b105a/license.rli /etc/license.rli
+  aws s3 cp s3://ha-test-bucket-3f5b105a/circle-installation-customizations /etc/circle-installation-customizations
+  curl https://get.replicated.com/docker | bash -s local_address=$(curl http://169.254.169.254/latest/meta-data/local-ipv4) no_proxy=1
+}
+
+time startup
+``` 
+
+The contents of the files pulled from S3 are as follows: 
+
+circle-installation-customizations:
+```
+# Note that connection strings below should be modified as necessary
+
+# Mongo DB
+MONGO_BASE_URI=mongodb://circle:<password>@<hostname>:27017
```

Comment:
If we assume a replica set then this should be something like 

```
MONGO_URI="mongodb://username:password@mongo1.ccie.sc-corp.net:27017,mongo2.ccie.sc-corp.net:27017,mongo3.ccie.sc-corp.net:27017"
```

@levlaz to update